.github/workflows/ci-pytorch-test-full.yml

```diff
@@ -20,84 +20,63 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, windows-2022, macOS-11]
-        python-version: ["3.7", "3.10"]  # minimum, maximum
-        requires: ["oldest", "latest"]
-        release: ["stable"]
-        exclude:
-          # There's no distribution of the oldest PyTorch 1.9 for Python 3.10.
-          # TODO: Remove the exclusion when dropping PyTorch 1.9 support.
-          - {python-version: "3.10", requires: "oldest"}
-        # TODO: re-enable RC testing
-        # include:
-        #   - {os: ubuntu-20.04, python-version: "3.10", requires: "latest", release: "pre"}
-
-    timeout-minutes: 40
+        include:
+          - {os: "ubuntu-20.04", python-version: "3.7", pytorch-version: "1.9", requires: "oldest"}
+          - {os: "ubuntu-20.04", python-version: "3.7", pytorch-version: "1.12"}
+          - {os: "ubuntu-20.04", python-version: "3.8", pytorch-version: "1.10"}
+          - {os: "ubuntu-20.04", python-version: "3.9", pytorch-version: "1.11"}
+          - {os: "ubuntu-20.04", python-version: "3.10", pytorch-version: "1.12"}
+          - {os: "macos-11", python-version: "3.7", pytorch-version: "1.9", requires: "oldest"}
+          - {os: "macos-11", python-version: "3.7", pytorch-version: "1.12"}
+          - {os: "macos-11", python-version: "3.8", pytorch-version: "1.10"}
+          - {os: "macos-11", python-version: "3.9", pytorch-version: "1.11"}
+          - {os: "macos-11", python-version: "3.10", pytorch-version: "1.12"}
+          - {os: "windows-2022", python-version: "3.7", pytorch-version: "1.9", requires: "oldest"}
+          - {os: "windows-2022", python-version: "3.7", pytorch-version: "1.12"}
+          - {os: "windows-2022", python-version: "3.8", pytorch-version: "1.10"}
+          - {os: "windows-2022", python-version: "3.9", pytorch-version: "1.11"}
+          - {os: "windows-2022", python-version: "3.10", pytorch-version: "1.12"}
```

Comment:
It is possible to have a more dense matrix like:
```yaml
matrix:
  os: [ubuntu-20.04, windows-2022, macOS-11]
  python-version: ["3.7", "3.8", "3.9", "3.10"]
  pytorch-version: ["1.9", "1.10", "1.11", "1.12"]
```
but IMO, I think these are too many to handle...